### PR TITLE
Implement timeout (typescript)

### LINKED ISSUE #189: API calls never timeouts when given invalid url
API should time out after a given time.

To reproduce:

    let iota = new IOTA({
        provider: 'http://example.com:14265',
    });
    iota.api.getNodeInfo((err, res) => { console.log(err ? err : res); }); // hangs forever

### PR DESCRIPTION
**Pull request by yet-another-account**
Same as #192 but for typescript. Fixes #189

### CODE REVIEW

**Comment by chrisdukakis** [comment]
#189

**Comment by eukaryote31** [comment]
Is there any easy way to add a timeout override option to every api call's options, and have the api one be the default for whenever it's unspecified? Or should I just add a setter like #192?

**Comment by chrisdukakis** [comment]
Hey @eukaryote31 thanks for working on this!
This is supported already, no action needed. Avoid setters & getters as much as possible. You can achieve same result automatically, without extra code. Create the functions you need like this:
```
const settings = { timeout: ... }
const getNodeInfo = createGetNodeInfo(provider(settings))
```

File: lib/utils/provider.ts [hunk 1]
```diff
@@ -40,7 +40,17 @@ export const provider = (settings?: Partial<Settings>): Provider => ({
             }
         }
 
-        return send(settings.provider || '', command)
+        let abortSignal = undefined
```
Comment (chrisdukakis) [inline lib/utils/provider.ts]: Please move this logic inside `send()` and pass to it the `timeout`. This way `send()` accepts `settings` only, and `batchedSend()` will automatically grab the timeout logic as well.

**Comment by chrisdukakis** [comment]
Lgtm! I'll just add a test for this soon.

**Comment by sachushaji** [comment]
Hi, Any updates on this. It would be nice if it was implemented

**Comment by obany** [comment]
Closing as these changes are on an unmaintained branch of the library, which will be incompatible with the new Chrysalis release of the network.

**Pull request closed without merging**